File: lysh-02.rb
#!/usr/bin/env ruby

require 'sqlite3'

search = ARGV.join(' ')
dbname = File.expand_path('~/Documents/Library.sqlite')
db = SQLite3::Database.open dbname

tables = db.execute("SELECT name FROM sqlite_master WHERE type='table'")
tables.flatten!
if tables.include? 'table_search'
  db.execute('DROP TABLE table_search')
end
db.execute('CREATE VIRTUAL TABLE table_search USING FTS5(Author,Editor,Translator,Title,Category,"ISBN-10","ISBN-13","DDC Number","LCC Number")')
db.execute('INSERT INTO table_search SELECT * FROM Sheet1')
results = db.execute('SELECT * FROM table_search WHERE table_search MATCH ?', search)
puts results

File: lysh-02.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Data::Dumper;
use DBI;
use feature 'say';

my $search = join(' ', @ARGV);
my $dbh = DBI->connect('DBI:SQLite:dbname=/home/michael/Documents/Library.sqlite');
my @tables = map {@$_} @{$dbh->selectall_arrayref("SELECT name FROM sqlite_master WHERE type='table'")};
if (grep('table_search', @tables))
{
    $dbh->do('DROP TABLE table_search');
}
$dbh->do('CREATE VIRTUAL TABLE table_search USING FTS5(Author,Editor,Translator,Title,Category,"ISBN-10","ISBN-13","DDC Number","LCC Number")');
my $sth = $dbh->prepare('INSERT INTO table_search SELECT * FROM Sheet1');
$sth->execute();
$sth = $dbh->prepare('SELECT * FROM table_search WHERE table_search MATCH ?');
$sth->execute($search);
my @results = @{$sth->fetchall_arrayref()};
say Dumper @results;
